File: python/signal_test.py
import os, signal, datetime, time
# signal.alarm(n) stop program in n second

def exit_script(signum, frame):
    print()
    print("exiting")
    exit(0)

def stop_script(signum, frame):
    print()
    print("I dont want to stop")

def teminate_script(signum, frame):
    print()
    print("I dont want to be terminated")

print("PID of process: ", os.getpid())

signal.signal(signal.SIGINT, exit_script) # ^C
signal.signal(signal.SIGTSTP, stop_script) # ^Z
signal.signal(signal.SIGTERM, teminate_script) # kill command

while True:
    now = datetime.datetime.now()
    current_time = now.strftime("%H:%M:%S")
    print(current_time)
    time.sleep(1)

File: cpp/signal_test.cpp
#include <csignal>
#include <iostream>
#include <chrono>
#include <thread>

void signal_handler(int signal) {
    std::cout << std::endl;
    std::cout << "Nah i dont want to exit." << std::endl;
}
 
int main() {
    std::cout << "PID of process: " << getpid() << std::endl;
    std::signal(SIGINT, signal_handler);

    int count = 1;
    while (1) {
        std::cout << count << std::endl;
        std::this_thread::sleep_for(std::chrono::milliseconds(1000));
        count++;
    }
}